### PR TITLE
Refactor test helpers and related types

### PR DESCRIPTION
**Pull request by narthur**

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -31,6 +32,28 @@ module.exports = {
 		jest: true,
 	},
 	rules: {
+		'@calm/react-intl/missing-formatted-message': [
+			2,
```
Comment (mattleff) [inline .eslintrc.js:36]: Should this (and below) be `error` or `warn`?

File: src/components/templates/andPlaybackContext.tsx [hunk 2]
```diff
@@ -165,10 +165,7 @@ export default function AndPlaybackContext({
 	const videoElRef = useRef<HTMLVideoElement>(null);
 	const originRef = useRef<HTMLDivElement>(null);
 
-	const [videojs, setVideojs] = useState<typeof VideoJs>();
-	useEffect(() => {
```
Comment (mattleff) [inline src/components/templates/andPlaybackContext.tsx:169]: Don't we still need this?

File: src/lib/getIntlMessages.ts [hunk 3]
```diff
@@ -4,8 +4,8 @@ import getLanguageByBaseUrl from '@lib/getLanguageByBaseUrl';
 
 export default function getIntlMessages(
 	languageRoute: string
-): Promise<ResolvedIntlConfig['messages']> {
+): ResolvedIntlConfig['messages'] {
 	const lang = getLanguageByBaseUrl(languageRoute, 'en');
 
-	return import(`../../public/compiled-lang/${lang?.base_url}.json`);
+	return lang?.messages || {};
```
Comment (mattleff) [inline src/lib/getIntlMessages.ts:10]: I think not using `import()` here will increase the bundle size.

File: src/lib/getNavigationItems.tsx [hunk 4]
```diff
@@ -394,6 +395,16 @@ export function getNavigationItems(
 				description: `Header nav link name: More`,
 			}),
 			children: [
+				{
```
Comment (mattleff) [inline src/lib/getNavigationItems.tsx:398]: AFAIK, this is for HelpScout.

File: tsconfig.json [hunk 5]
```diff
@@ -26,7 +26,8 @@
 			"@lib/*": ["src/lib/*"],
 			"@components/*": ["src/components/*"],
 			"@containers/*": ["src/containers/*"],
-			"@pages/*": ["src/pages/*"]
+			"@pages/*": ["src/pages/*"],
+			"@root/*": ["*"]
```
Comment (mattleff) [inline tsconfig.json:30]: Is this change needed?

**Review by mattleff — CHANGES_REQUESTED** [review]
Looks good overall! It seems the checks have stopped running and I think they would catch a few issues I noted in the comments. I also think the additional langs need a `npm run extract && npm run compile`.

**Comment by mattleff** [comment]
@narthur The changes in this PR have significant bundle implications. I've highlighted a couple of them in my comments.

File: src/components/templates/andPlaybackContext.tsx [hunk 6]
```diff
@@ -400,20 +397,17 @@ export default function AndPlaybackContext({
 				defaultVolume: 1,
 				sources,
 			};
+
 			if (playerRef.current) {
 				playerRef.current.src(sources);
 				resetPlayer();
-			} else if (videojs) {
```
Comment (mattleff) [inline src/components/templates/andPlaybackContext.tsx:406]: I believe this reintroduces the bug #225 fixed.

File: tsconfig.json (hunk 5 above)
Comment (narthur) [reply tsconfig.json:30]: Good catch, resolved!

**Comment by narthur** [comment]
@mattleff I just ran `npm run extract && npm run compile` and it didn't result in any changes... 😕

File: src/components/templates/andPlaybackContext.tsx (hunk 6 above)
Comment (mattleff) [reply src/components/templates/andPlaybackContext.tsx:406]: @narthur I can reproduce the bug in Safari: 

1. In a Safari incognito window (or anywhere with uncached JS) open the site.
2. Click play on one of the home page recording cards.

### Expected result

The recording plays. This is the current behavior on the public website.

### Actual result

The recording loads the miniplayer but doesn't start playing.

**Review by mattleff — CHANGES_REQUESTED** [review]
Looks great, except for the Safari two-click bug

**Review by mattleff — APPROVED** [review]
Can't reproduce the error anymore... Let's ship it and see if we get complaints.

**Pull request merged**